src/cheshire/core.clj

```diff
@@ -24,9 +49,12 @@
          generator (.createGenerator
                     ^JsonFactory (or factory/*json-factory*
                                      factory/json-factory)
-                    ^Writer sw)]
-     (when (:pretty opt-map)
+                    ^Writer sw)
+         print-pretty (:pretty opt-map)]
+     (when (= true print-pretty)
```

Comment:
I'm not sure of the performance impact, but it might be nice to hide all the type comparisons behind a check and compare benchmarks, something like:

```clj
(when print-pretty
  (condp instance? print-pretty
    java.lang.Boolean
    (.useDefaultPrettyPrinter generator)
    clojure.lang.IPersistentMap
    (.setPrettyPrinter generator (create-pretty-printer print-pretty))
    org.codehaus.jackson.PrettyPrinter
    (.setPrettyPrinter generator print-pretty)
```